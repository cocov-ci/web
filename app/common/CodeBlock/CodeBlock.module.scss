@use "sass:map";
@use "sass:color";
@import "styles/variables.module";

.base {
  background: map.get($colorBox, "background");
  border: 1px solid map.get($colorBox, "border");
  border-radius: 8px;
  overflow: scroll;

  .table {
    border-collapse: collapse;

    tbody {
      tr:first-child td {
        padding-top: 10px;
      }

      tr:last-child td {
        padding-bottom: 10px;
      }
    }

    .line {
      pre {
        font-size: 16px;
        letter-spacing: 0;
        text-align: left;
      }
    }

    .warningLine {
      padding-top: 3px;
      pre { display: inline-block; }

      .warning {
        background-image: linear-gradient(180deg, #f7c008 0%, #e3b107 100%);
        border: 1px solid #a88305;
        border-radius: 4px;
        text-align: left;
        padding: 3px 12px 3px 6px;
        display: inline-flex;
        box-shadow: 0 2px 7px 0 color.adjust($colorBlack, $alpha: -0.78);
        min-width: 360px;
        max-width: 520px;
        margin-bottom: 10px;
        text-shadow: 0 0 1px #c2c814;

        .icon {
          flex-shrink: 0;
          margin-right: 6px;
        }

        .content {
          line-height: 19px;
          font-size: 14px;
          text-shadow: 0 0 1px #b2b559;
        }
      }
    }

    .gutter {
      user-select: none;
      border-right: 1px solid #979797;
      text-align: right;

      div {
        padding: 0 10px;
        font-size: 16px;
        color: #838383;
      }
    }
  }
}
