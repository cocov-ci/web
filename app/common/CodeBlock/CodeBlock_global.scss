.codeblock-line {
  span {
    &.k,
    &.kp,
    &.kn,
    &.kd,
    &.kt,
    &.kc { color: #025bb7; }

    &.nf { color: rgb(55 113 114); }

    &.na { color: #1b7878; }

    &.n { color: rgb(59 59 59); }

    &.nb,
    &.vi,
    &.nd,
    &.no,
    &.nl { color: #8739ac; }

    &.nc,
    &.nn {
      color: rgb(72 151 180);
      font-weight: bold;
    }

    &.s,
    &.ss,
    &.s2,
    &.sx,
    &.sr { color: rgb(185 57 28); }

    &.nb + .o + .no { color: black; }

    &.m,
    &.mi,
    &.mh,
    &.mf,
    &.mo,
    &.mb { color: rgb(197 70 108); }

    &.o,
    &.p { color: rgb(119 126 136); }

    &.c1 {
      color: rgb(119 119 119);
      font-style: italic;
    }
  }
}
