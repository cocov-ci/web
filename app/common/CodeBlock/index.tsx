import classNames from 'classnames'
import { AlertTriangle } from 'lucide-react'
import React from 'react'

import { inconsolata } from 'utils/fonts'

import styles from './CodeBlock.module.scss'

type LineProps = {
  line: number
  source: string
}
type WarningProps = {
  text: string
  padding: string
}

type CodeBlockProps = {
  className?: string
  objects?: [any]
}

const Line = ({ line, source }: LineProps) => {
  return (
    <tr>
      <td className={classNames(styles.gutter, inconsolata.className)}>
        <div>{line}</div>
      </td>
      <td
        className={classNames(
          'codeblock-line',
          styles.line,
          inconsolata.className,
        )}
        dangerouslySetInnerHTML={{ __html: source }}
      />
    </tr>
  )
}

const Warning = ({ text, padding }: WarningProps) => {
  return (
    <tr>
      <td className={styles.gutter} />
      <td className={classNames(styles.warningLine)}>
        <pre>{padding}</pre>
        <div className={classNames(styles.warning)}>
          <AlertTriangle className={classNames(styles.icon)} size={20} />
          <div className={classNames(styles.content)}>{text}</div>
        </div>
      </td>
    </tr>
  )
}

const CodeBlock = ({ className, objects }: CodeBlockProps) => {
  return (
    <div className={classNames(styles.base, className)}>
      <table className={classNames(styles.table)}>
        <tbody>
          {objects?.map(obj => {
            if (obj?.source) {
              return (
                <Line
                  key={`line=${obj.line}`}
                  line={obj.line}
                  source={obj.source}
                />
              )
            } else {
              return (
                <Warning key="warning" padding={obj.padding} text={obj.text} />
              )
            }
          })}
        </tbody>
      </table>
    </div>
  )
}

export default CodeBlock
