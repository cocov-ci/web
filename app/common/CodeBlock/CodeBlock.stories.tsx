import type { Meta, StoryObj } from '@storybook/react'

import CodeBlock from './index'

const meta: Meta<typeof CodeBlock> = {
  title: 'Common/CodeBlock',
  component: CodeBlock,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof CodeBlock>

export const Default: Story = {
  args: {
    issue: {
      objects: [
        {
          kind: 'line',
          line: 1,
          source:
            '<pre>    <span class="k">def</span> <span class="nf">repo_path</span><span class="p">(</span><span class="n">repo</span><span class="p">)</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 2,
          source:
            '<pre>      <span class="n">base</span> <span class="o">=</span> <span class="n">base_path</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 3,
          source:
            '<pre>      <span class="nb">name</span> <span class="o">=</span> <span class="no">Digest</span><span class="o">::</span><span class="no">SHA1</span><span class="p">.</span><span class="nf">hexdigest</span><span class="p">(</span><span class="n">repo</span><span class="p">)</span>\n</pre>',
        },
        {
          kind: 'warn',
          text: 'Weak hashing algorithm used: SHA1',
          padding: '      ',
        },
        {
          kind: 'line',
          line: 4,
          source:
            '<pre>      <span class="k">if</span> <span class="n">base</span><span class="p">.</span><span class="nf">is_a?</span> <span class="no">Pathname</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 5,
          source:
            '<pre>        <span class="n">base</span><span class="p">.</span><span class="nf">join</span><span class="p">(</span><span class="nb">name</span><span class="p">)</span>\n</pre>',
        },
      ],
    },
  },
}

export const LongMessage: Story = {
  args: {
    issue: [
      {
        type: 'line',
        line: 1,
        source:
          '<pre>    <span class="k">def</span> <span class="nf">repo_path</span><span class="p">(</span><span class="n">repo</span><span class="p">)</span>\n</pre>',
      },
      {
        type: 'line',
        line: 2,
        source:
          '<pre>      <span class="n">base</span> <span class="o">=</span> <span class="n">base_path</span>\n</pre>',
      },
      {
        type: 'line',
        line: 3,
        source:
          '<pre>      <span class="nb">name</span> <span class="o">=</span> <span class="no">Digest</span><span class="o">::</span><span class="no">SHA1</span><span class="p">.</span><span class="nf">hexdigest</span><span class="p">(</span><span class="n">repo</span><span class="p">)</span>\n</pre>',
      },
      {
        type: 'warn',
        text: '$METHOD is a state-changing MVC method that does not validate the antiforgery token or do strict content-type checking. State-changing controller methods should either enforce antiforgery tokens or do strict content-type checking to prevent simple HTTP request types from bypassing CORS preflight controls.',
        padding: '      ',
      },
      {
        type: 'line',
        line: 4,
        source:
          '<pre>      <span class="k">if</span> <span class="n">base</span><span class="p">.</span><span class="nf">is_a?</span> <span class="no">Pathname</span>\n</pre>',
      },
      {
        type: 'line',
        line: 5,
        source:
          '<pre>        <span class="n">base</span><span class="p">.</span><span class="nf">join</span><span class="p">(</span><span class="nb">name</span><span class="p">)</span>\n</pre>',
      },
    ],
  },
}

export const Coverage: Story = {
  args: {
    coverage: {
      blocks: [
        { kind: 'neutral', start: 1, end: 2 },
        { kind: 'covered', start: 3, end: 7 },
        { kind: 'neutral', start: 8, end: 8 },
        { kind: 'covered', start: 9, end: 12 },
        { kind: 'neutral', start: 13, end: 14 },
        { kind: 'covered', start: 15, end: 15 },
        { kind: 'neutral', start: 16, end: 16 },
        { kind: 'covered', start: 17, end: 19 },
        { kind: 'neutral', start: 20, end: 20 },
        { kind: 'covered', start: 21, end: 21 },
        { kind: 'neutral', start: 22, end: 22 },
        { kind: 'covered', start: 23, end: 24 },
        { kind: 'neutral', start: 25, end: 26 },
        { kind: 'covered', start: 27, end: 29 },
        { kind: 'neutral', start: 30, end: 30 },
        { kind: 'covered', start: 31, end: 33 },
        { kind: 'neutral', start: 34, end: 34 },
        { kind: 'covered', start: 35, end: 35 },
        { kind: 'neutral', start: 36, end: 36 },
        { kind: 'covered', start: 37, end: 37 },
        { kind: 'neutral', start: 38, end: 40 },
        { kind: 'covered', start: 41, end: 42 },
        { kind: 'neutral', start: 43, end: 43 },
        { kind: 'missed', start: 44, end: 45 },
        { kind: 'neutral', start: 46, end: 46 },
        { kind: 'missed', start: 47, end: 47 },
        { kind: 'neutral', start: 48, end: 50 },
      ],
      source: [
        {
          kind: 'line',
          line: 1,
          source:
            '<pre><span class="c1"># frozen_string_literal: true</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 2,
          source: '<pre></pre>',
        },
        {
          kind: 'line',
          line: 3,
          source:
            '<pre><span class="k">module</span> <span class="nn">V1</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 4,
          source:
            '<pre>  <span class="k">class</span> <span class="nc">GithubEventsController</span> <span class="o">&lt;</span> <span class="no">V1Controller</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 5,
          source:
            '<pre>    <span class="n">before_action</span> <span class="ss">:validate_signature</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 6,
          source:
            '<pre>    <span class="n">before_action</span> <span class="ss">:validate_event</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 7,
          source:
            '<pre>    <span class="n">around_action</span> <span class="ss">:ignore_duplicated_events</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 8,
          source: '<pre></pre>',
        },
        {
          kind: 'line',
          line: 9,
          source:
            '<pre>    <span class="k">def</span> <span class="nf">create</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 10,
          source:
            '<pre>      <span class="n">key</span> <span class="o">=</span> <span class="p">[</span><span class="vi">@event_name</span><span class="p">,</span> <span class="vi">@event</span><span class="p">[</span><span class="ss">:action</span><span class="p">]</span><span class="o">&amp;</span><span class="p">.</span><span class="nf">to_sym</span><span class="p">].</span><span class="nf">compact</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 11,
          source:
            '<pre>      <span class="no">WebhookProcessorService</span><span class="p">.</span><span class="nf">call</span><span class="p">(</span><span class="n">key</span><span class="p">,</span> <span class="vi">@event</span><span class="p">)</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 12,
          source:
            '<pre>      <span class="n">head</span> <span class="ss">:ok</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 13,
          source: '<pre>    <span class="k">end</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 14,
          source: '<pre></pre>',
        },
        {
          kind: 'line',
          line: 15,
          source: '<pre>    <span class="kp">private</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 16,
          source: '<pre></pre>',
        },
        {
          kind: 'line',
          line: 17,
          source:
            '<pre>    <span class="k">def</span> <span class="nf">validate_event</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 18,
          source:
            '<pre>      <span class="vi">@event_name</span> <span class="o">=</span> <span class="n">request</span><span class="p">.</span><span class="nf">env</span><span class="p">[</span><span class="s2">"HTTP_X_GITHUB_EVENT"</span><span class="p">]</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 19,
          source:
            '<pre>      <span class="k">return</span> <span class="n">head</span> <span class="ss">:bad_request</span> <span class="k">if</span> <span class="vi">@event_name</span><span class="p">.</span><span class="nf">blank?</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 20,
          source: '<pre></pre>',
        },
        {
          kind: 'line',
          line: 21,
          source:
            '<pre>      <span class="vi">@event_name</span> <span class="o">=</span> <span class="vi">@event_name</span><span class="p">.</span><span class="nf">to_sym</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 22,
          source: '<pre></pre>',
        },
        {
          kind: 'line',
          line: 23,
          source:
            '<pre>      <span class="n">request</span><span class="p">.</span><span class="nf">body</span><span class="p">.</span><span class="nf">rewind</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 24,
          source:
            '<pre>      <span class="vi">@event</span> <span class="o">=</span> <span class="no">JSON</span><span class="p">.</span><span class="nf">parse</span><span class="p">(</span><span class="n">request</span><span class="p">.</span><span class="nf">body</span><span class="p">.</span><span class="nf">read</span><span class="p">,</span> <span class="ss">symbolize_names: </span><span class="kp">true</span><span class="p">).</span><span class="nf">with_indifferent_access</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 25,
          source: '<pre>    <span class="k">end</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 26,
          source: '<pre></pre>',
        },
        {
          kind: 'line',
          line: 27,
          source:
            '<pre>    <span class="k">def</span> <span class="nf">ignore_duplicated_events</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 28,
          source:
            '<pre>      <span class="n">event_id</span> <span class="o">=</span> <span class="n">request</span><span class="p">.</span><span class="nf">env</span><span class="p">[</span><span class="s2">"HTTP_X_GITHUB_DELIVERY"</span><span class="p">]</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 29,
          source:
            '<pre>      <span class="k">return</span> <span class="n">head</span> <span class="ss">:bad_request</span> <span class="k">if</span> <span class="n">event_id</span><span class="p">.</span><span class="nf">blank?</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 30,
          source: '<pre></pre>',
        },
        {
          kind: 'line',
          line: 31,
          source:
            '<pre>      <span class="n">event_key</span> <span class="o">=</span> <span class="s2">"github:event:delivery:</span><span class="si">#{</span><span class="n">event_id</span><span class="si">}</span><span class="s2">"</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 32,
          source:
            '<pre>      <span class="no">Cocov</span><span class="o">::</span><span class="no">Redis</span><span class="p">.</span><span class="nf">lock</span><span class="p">(</span><span class="s2">"github:delivery:</span><span class="si">#{</span><span class="n">event_id</span><span class="si">}</span><span class="s2">"</span><span class="p">,</span> <span class="mi">1</span><span class="p">.</span><span class="nf">minute</span><span class="p">)</span> <span class="k">do</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 33,
          source:
            '<pre>        <span class="n">head</span> <span class="ss">:ok</span> <span class="n">and</span> <span class="k">return</span> <span class="k">if</span> <span class="no">Cocov</span><span class="o">::</span><span class="no">Redis</span><span class="p">.</span><span class="nf">instance</span><span class="p">.</span><span class="nf">exists?</span> <span class="n">event_key</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 34,
          source: '<pre></pre>',
        },
        {
          kind: 'line',
          line: 35,
          source: '<pre>        <span class="k">yield</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 36,
          source: '<pre></pre>',
        },
        {
          kind: 'line',
          line: 37,
          source:
            '<pre>        <span class="no">Cocov</span><span class="o">::</span><span class="no">Redis</span><span class="p">.</span><span class="nf">instance</span><span class="p">.</span><span class="nf">set</span> <span class="n">event_key</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="ss">ex: </span><span class="mi">2</span><span class="p">.</span><span class="nf">days</span> <span class="k">if</span> <span class="n">response</span><span class="p">.</span><span class="nf">successful?</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 38,
          source: '<pre>      <span class="k">end</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 39,
          source: '<pre>    <span class="k">end</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 40,
          source: '<pre></pre>',
        },
        {
          kind: 'line',
          line: 41,
          source:
            '<pre>    <span class="k">def</span> <span class="nf">validate_signature</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 42,
          source:
            '<pre>      <span class="k">return</span> <span class="k">if</span> <span class="no">Cocov</span><span class="o">::</span><span class="no">GITHUB_WEBHOOK_SECRET_KEY</span><span class="p">.</span><span class="nf">blank?</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 43,
          source: '<pre></pre>',
        },
        {
          kind: 'line',
          line: 44,
          source:
            '<pre>      <span class="n">request</span><span class="p">.</span><span class="nf">body</span><span class="p">.</span><span class="nf">rewind</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 45,
          source:
            '<pre>      <span class="n">hex_digest</span> <span class="o">=</span> <span class="no">OpenSSL</span><span class="o">::</span><span class="no">HMAC</span><span class="p">.</span><span class="nf">hexdigest</span><span class="p">(</span><span class="no">OpenSSL</span><span class="o">::</span><span class="no">Digest</span><span class="p">.</span><span class="nf">new</span><span class="p">(</span><span class="s2">"sha256"</span><span class="p">),</span> <span class="no">Cocov</span><span class="o">::</span><span class="no">GITHUB_WEBHOOK_SECRET_KEY</span><span class="p">,</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 46,
          source:
            '<pre>        <span class="n">request</span><span class="p">.</span><span class="nf">body</span><span class="p">.</span><span class="nf">read</span><span class="p">)</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 47,
          source:
            '<pre>      <span class="n">head</span> <span class="ss">:forbidden</span> <span class="k">unless</span> <span class="no">Rack</span><span class="o">::</span><span class="no">Utils</span><span class="p">.</span><span class="nf">secure_compare</span><span class="p">(</span><span class="s2">"sha256=</span><span class="si">#{</span><span class="n">hex_digest</span><span class="si">}</span><span class="s2">"</span><span class="p">,</span> <span class="n">request</span><span class="p">.</span><span class="nf">env</span><span class="p">[</span><span class="s2">"HTTP_X_HUB_SIGNATURE_256"</span><span class="p">])</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 48,
          source: '<pre>    <span class="k">end</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 49,
          source: '<pre>  <span class="k">end</span>\n</pre>',
        },
        {
          kind: 'line',
          line: 50,
          source: '<pre><span class="k">end</span>\n</pre>',
        },
      ],
    },
  },
}
