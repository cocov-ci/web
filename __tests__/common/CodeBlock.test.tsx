import { render, screen } from '@testing-library/react'

import CodeBlock from 'app/common/CodeBlock'

import '@testing-library/jest-dom'

const issue = [
  {
    type: 'line' as const,
    line: 1,
    source:
      '<pre>    <span class="k">def</span> <span class="nf">repo_path</span><span class="p">(</span><span class="n">repo</span><span class="p">)</span>\n</pre>',
  },
  {
    type: 'warn' as const,
    text: 'Weak hashing algorithm used: SHA1',
    padding: '      ',
  },
  {
    type: 'line' as const,
    line: 2,
    source:
      '<pre>      <span class="n">base</span> <span class="o">=</span> <span class="n">base_path</span>\n</pre>',
  },
]

const coverage = {
  blocks: [
    { kind: 'neutral' as const, start: 1, end: 2 },
    { kind: 'covered' as const, start: 3, end: 7 },
    { kind: 'neutral' as const, start: 8, end: 8 },
  ],
  source: [
    {
      kind: 'line' as const,
      line: 1,
      source:
        '<pre><span class="c1"># frozen_string_literal: true</span>\n</pre>',
    },
    {
      kind: 'line' as const,
      line: 2,
      source: '<pre></pre>',
    },
  ],
}

const plainText =
  "Execution failed. Plugin exited with status 1:\nFailed processing: uninitialized constant PluginKit\n\n  rescue PluginKit::Exec::ExecutionError =\u003e e\n         ^^^^^^^^^\n/plugin.rb:4:in `rescue in block in \u003ctop (required)\u003e'\n/plugin.rb:2:in `block in \u003ctop (required)\u003e'\n/home/cocov/.gem/gems/cocov_plugin_kit-0.1.6/lib/cocov/plugin_kit.rb:41:in `instance_exec'\n/home/cocov/.gem/gems/cocov_plugin_kit-0.1.6/lib/cocov/plugin_kit.rb:41:in `block in run'\n/home/cocov/.gem/gems/cocov_plugin_kit-0.1.6/lib/cocov/plugin_kit.rb:38:in `chdir'\n/home/cocov/.gem/gems/cocov_plugin_kit-0.1.6/lib/cocov/plugin_kit.rb:38:in `run'\n/plugin.rb:1:in `\u003ctop (required)\u003e'\n\u003cinternal:/usr/local/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb\u003e:85:in `require'\n\u003cinternal:/usr/local/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb\u003e:85:in `require'\n/home/cocov/.gem/gems/cocov_plugin_kit-0.1.6/exe/cocov:11:in `\u003ctop (required)\u003e'\n/home/cocov/.gem/bin/cocov:25:in `load'\n/home/cocov/.gem/bin/cocov:25:in `\u003cmain\u003e'\n"

describe('common/CodeBlock', () => {
  it('renders CodeBlock with "issue" param', () => {
    render(<CodeBlock issue={issue} />)

    expect(screen.getByText('repo_path')).toBeVisible()
  })

  it('renders CodeBlock with "coverage" param', () => {
    render(<CodeBlock coverage={coverage} />)

    expect(screen.getByText('# frozen_string_literal: true')).toBeVisible()
  })

  it('renders CodeBlock with "coverage" param', () => {
    render(<CodeBlock plainText={plainText} />)

    expect(
      screen.getByText('Execution failed. Plugin exited with status 1:'),
    ).toBeVisible()
  })

  it('renders CodeBlock snapshots', () => {
    const { container } = render(<CodeBlock issue={issue} />)
    expect(container).toMatchSnapshot()
  })
})
