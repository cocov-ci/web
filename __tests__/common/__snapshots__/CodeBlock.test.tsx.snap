// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/CodeBlock renders CodeBlock snapshots 1`] = `
<div>
  <div
    class="base"
  >
    <table
      class="table"
    >
      <tbody>
        <tr>
          <td
            class="gutter className"
          >
            <div>
              1
            </div>
          </td>
          <td
            class="codeblock-line line className"
          >
            <div>
              <pre>
                    
                <span
                  class="k"
                >
                  def
                </span>
                 
                <span
                  class="nf"
                >
                  repo_path
                </span>
                <span
                  class="p"
                >
                  (
                </span>
                <span
                  class="n"
                >
                  repo
                </span>
                <span
                  class="p"
                >
                  )
                </span>
                

              </pre>
            </div>
          </td>
        </tr>
        <tr>
          <td
            class="gutter"
          />
          <td
            class="warningLine"
          >
            <pre>
                    
            </pre>
            <div
              class="warning"
            >
              <svg
                class="icon"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m21.73 18-8-14a2 2 0 0 0-3.48 0l-8 14A2 2 0 0 0 4 21h16a2 2 0 0 0 1.73-3Z"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="9"
                  y2="13"
                />
                <line
                  x1="12"
                  x2="12.01"
                  y1="17"
                  y2="17"
                />
              </svg>
              <div
                class="content"
              >
                Weak hashing algorithm used: SHA1
              </div>
            </div>
          </td>
        </tr>
        <tr>
          <td
            class="gutter className"
          >
            <div>
              2
            </div>
          </td>
          <td
            class="codeblock-line line className"
          >
            <div>
              <pre>
                      
                <span
                  class="n"
                >
                  base
                </span>
                 
                <span
                  class="o"
                >
                  =
                </span>
                 
                <span
                  class="n"
                >
                  base_path
                </span>
                

              </pre>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
